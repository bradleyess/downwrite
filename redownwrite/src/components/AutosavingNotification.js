import React from 'react'
import Toast from './Toast'

/* TypeError: cyclic object value */
export default () => <Toast>Autosaving</Toast>
