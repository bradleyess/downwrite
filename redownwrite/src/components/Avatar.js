import React, { Component } from 'react'
import styled from 'styled-components'
import color from 'rcolor'

const initialState = {
  a: color() || `#FEB692`,
  b: color() || `#EA5455`
}

class RandomColor extends Component {
  state = initialState

  render() {
    return this.props.children(this.state)
  }
}

const Avatar = styled.div`
  border-radius: 3rem;
  height: 3rem;
  width: 3rem;
  background: linear-gradient(
    135deg,
    ${props => props.colors.a || '#FEB692'} 10%,
    ${props => props.colors.b || '#EA5455'} 100%
  );
  margin: 0 auto 1rem;
`

export default () => <RandomColor>{colors => <Avatar colors={colors} />}</RandomColor>
