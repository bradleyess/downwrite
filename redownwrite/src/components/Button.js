// @flow

import styled from 'styled-components'
import { colors } from '../utils/defaultStyles'

export default styled.button`
  display: block;
  font-size: 14px;
  background-color: ${colors.yellow700};
  transition: background-color 250ms ease-in-out;
  color: #282828;
  font-weight: 700;
  border: 0;
  padding: 0.25rem 1.125rem;
  border-radius: 0.25rem;
  box-shadow: 0 0 4px rgba(0, 0, 0, 0.037), 0 4px 8px rgba(0, 0, 0, 0.07);
  &:hover {
    background-color: ${colors.yellow500};
  }

  &:[disabled] {
    filter: grayscale(100%);
  }
`
