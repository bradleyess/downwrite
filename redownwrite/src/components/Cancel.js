import styled from 'styled-components'
import { colors } from '../utils/defaultStyles'

export default styled.button`
  margin-right: 1rem;
  color: ${colors.blue700};
  background: none;
  border: 0px;
  font-weight: 700;
`
