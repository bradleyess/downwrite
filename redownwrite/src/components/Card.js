import React from 'react'
import styled from 'styled-components'
import Link from 'react-router-dom/Link'
import distance from 'date-fns/distance_in_words_to_now'
import { colors } from '../utils/defaultStyles'

const CardContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 192px;
  font-weight: 400;
  box-shadow: 0 0 2px rgba(0, 0, 0, 0.07), 0 2px 4px rgba(0, 0, 0, 0.12);
  background-color: white;
`

const CardTitle = styled.h2`
  font-size: 14px;
  margin-bottom: 0px;
  font-weight: 700;

  @media (min-width: 57.75rem) {
    font-size: 18px;
  }
`

const CardHeader = styled.header`
  display: block;
  border-bottom: 1px solid ${colors.gray200};
  padding: 12px 8px;
`

const CardMeta = styled.small`
  font-size: 12px;
  display: block;
  color: ${colors.gray300};
  font-weight: 400;
`

const CardTray = styled.footer`
  display: flex;
  padding: 8px;
  font-weight: 700;
  font-size: 12px;
  justify-content: space-between;
  background-color: rgba(101, 163, 191, 0.125);

  & a:hover {
    color: ${colors.blue500};
  }
`

const CardPreview = styled.p`
  font-size: 12px;
`

const CardBody = styled.div`
  display: block;
  padding: 0.5rem;
  flex: 1;
  overflow: hidden;
`

const CardDelete = styled.button`
  color: ${colors.blue700};
  border: 0px;
  background: none;
  appearance: none;
  -webkit-font-smoothing: antialiased;
`

const SxLink = styled(Link)`
  margin-right: 8px;
`

const Card = ({ title, id, content, dateAdded, onDelete, ...args }) => (
  <CardContainer data-testid="CARD">
    <CardHeader>
      <CardTitle data-testid="CARD_TITLE">
        <Link to={`/${id}/edit`}>{title}</Link>
      </CardTitle>
      <CardMeta>added {distance(dateAdded)} ago</CardMeta>
    </CardHeader>

    <CardBody>
      {content && (
        <CardPreview data-test="snippet">{content.blocks[0].text.substr(0, 75)}</CardPreview>
      )}
    </CardBody>
    <CardTray>
      <div data-testid="CARD_EXCERPT">
        <SxLink to={`/${id}/edit`}>Edit</SxLink>
        {args.public && <Link to={`/${id}/preview`}>Preview</Link>}
      </div>
      {onDelete && (
        <CardDelete data-testid="CARD_DELETE_BUTTON" onClick={onDelete}>
          Delete
        </CardDelete>
      )}
    </CardTray>
  </CardContainer>
)

export default Card
