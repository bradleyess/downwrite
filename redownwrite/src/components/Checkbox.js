import React from 'react'
import { css } from 'glamor'
import icon from './check.svg'

const sharedStyle = css({
  display: 'inline-block',
  borderRadius: 4,
  verticalAlign: 'middle',
  width: 20,
  height: 20,
  appearance: 'none',
  border: 0
})

const checkedStyle = css(sharedStyle, {
  background: `var(--color-1) url(${icon}) no-repeat center center`,
  border: 0
})

const unCheckedStyle = css(sharedStyle, { background: '#D0D0D0' })

export default props => (
  <input
    type="checkbox"
    className={css(props.checked ? checkedStyle : unCheckedStyle)}
    {...props}
  />
)
