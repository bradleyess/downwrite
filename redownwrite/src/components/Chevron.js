import React from 'react'

export default ({ open }: { open: boolean }) => (
  <svg
    className="Chevron"
    viewBox="0 0 16 16"
    width="16"
    height="16"
    fill="none"
    stroke="currentColor"
    strokeWidth={2}>
    <path
      d={
        open
          ? 'M1.0606601717798212 11 L8 4.060660171779821 L14.939339828220179 11'
          : 'M1.0606601717798212 5 L8 11.939339828220179 L14.939339828220179 5'
      }
    />
  </svg>
)
