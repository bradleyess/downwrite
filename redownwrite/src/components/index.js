export { default as Button } from './Button'
export { default as Cancel } from './Cancel'
export { default as Card } from './Card'
export { default as DWEditor } from './Draft'
export { default as NightMode } from './NightMode'
export { default as EmptyPosts } from './EmptyPosts'
export { default as Header } from './Header'
export { default as Input } from './Input'
export { default as Loading } from './Loading'
export { default as Logo } from './Logo'
export { default as Nav } from './Nav'
export { default as PostList } from './PostList'
export { default as Toolbar } from './Toolbar'
export { default as Wrapper } from './Wrapper'
export { default as Helpers } from './Helpers'
export { default as LoginInput } from './LoginInput'
export { default as LoginButton } from './LoginButton'
export { default as SettingsIcon } from './SettingsIcon'
export { default as Layout } from './Layout'
export { default as Modal } from './Modal'
export { default as Toggle } from './Toggle'
export { default as Export } from './Export'
export { default as Privacy } from './Privacy'
export { default as Check } from './Checkbox'
export { default as CloseIcon } from './CloseIcon'
export { default as InvalidToken } from './InvalidToken'
export { default as Content } from './Content'
export { default as Logger } from './Logger'
export { default as UIFlash } from './UIFlashMessages'
