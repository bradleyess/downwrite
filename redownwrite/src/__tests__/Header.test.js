import { Header } from '../components'

let wrapper = mount(
  <MemoryRouter>
    <Header authed name="Downwrite" />
  </MemoryRouter>
)

describe('Header Component', () => {
  it('renders a header', () => {
    expect(wrapper.exists()).toBe(true)
  })

  it('contains application name', () => {
    expect(wrapper.find('h1').contains('Downwrite')).toBe(true)
  })
})
