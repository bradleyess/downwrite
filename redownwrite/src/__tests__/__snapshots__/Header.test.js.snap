// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches snapshot 1`] = `
<header
  class="sc-fBuWsC gUQLbY"
  data-testid="APP_HEADER"
>
  <nav
    class="sc-csuQGl fUdZHz"
  >
    <div
      class="sc-gipzik eaVZiQ"
    >
      <svg
        width="33"
        height="50"
        viewBox="0 0 33 50"
      >
        <title>
          Downwrite Logo
        </title>
        <defs>
          <radialgradient
            fx="50%"
            fy="50%"
            r="143.827%"
            id="a"
          >
            <stop
              stop-color="#FFCE00"
              offset="0%"
            />
            <stop
              stop-color="#FFD600"
              offset="16.39%"
            />
            <stop
              stop-color="#FFE600"
              offset="57.9%"
            />
            <stop
              stop-color="#FFCE00"
              offset="100%"
            />
          </radialgradient>
          <radialgradient
            fx="50%"
            fy="50%"
            r="143.827%"
            id="b"
          >
            <stop
              stop-color="#FFB600"
              offset="0%"
            />
            <stop
              stop-color="#FFC200"
              offset="52.49%"
            />
            <stop
              stop-color="#F4A903"
              offset="100%"
            />
          </radialgradient>
        </defs>
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <path
            d="M16.469 5.275V30.97c1.202 0 2.176.965 2.176 2.155s-.974 2.155-2.176 2.155v9.444l10.101-9.594L16.47 5.275z"
            fill="url(#a)"
          />
          <path
            d="M14.293 33.126c0-1.19.974-2.155 2.176-2.155V5.275L6.367 35.13l10.102 9.594v-9.444a2.165 2.165 0 0 1-2.176-2.155z"
            fill="url(#b)"
          />
          <path
            d="M32.938 50H0V0h32.938v50zM1.63 48.47h29.677V1.53H1.63v46.94z"
            fill="#FFC200"
          />
        </g>
      </svg>
    </div>
    <a
      class="sc-bRBYWo isFbr"
      href="/"
    >
      <h1
        class="sc-jhAzac dVEBXt"
        data-testid="APP_HEADER_TITLE"
      >
        Downwrite
      </h1>
    </a>
  </nav>
  <nav
    class="sc-csuQGl fUdZHz"
  >
    <a
      class="sc-Rmtcm jYTvmM"
      href="/new"
    >
      New
    </a>
    <button
      class="sc-hzDkRC fAgjdi"
    >
      <svg
        width="20px"
        height="9px"
        viewBox="0 0 20 9"
      >
        <desc>
          Navicon
        </desc>
        <g
          stroke="none"
          stroke-width="1"
          fill="none"
          fill-rule="evenodd"
        >
          <g
            fill="#4C4C4C"
          >
            <rect
              id="Rectangle-Copy-3"
              x="0"
              y="0"
              width="20"
              height="1"
            />
            <rect
              id="Rectangle-Copy-4"
              x="10"
              y="4"
              width="10"
              height="1"
            />
            <rect
              id="Rectangle-Copy-5"
              x="5"
              y="8"
              width="15"
              height="1"
            />
          </g>
        </g>
      </svg>
    </button>
  </nav>
</header>
`;
